### ISSUE TITLE
Will onnx support dcn future?

### ISSUE THREAD

**Issue by jinfagang** [onnx/onnx#2395]
Is that deformable convolution network will be added into onnx?

**Comment by ebarsoum** [comment]
Is DCN implemented in any framework natively?

**Comment by jinfagang** [comment]
I think not. However, this is a good feature since DCN makes convolution networks gain a massive accuracy. I'll go to framework community to ask this feature

**Comment by ebarsoum** [comment]
But to generate it in ONNX it need to be converted from someplace. If this cause a massive accuracy gain, why no framework implemented it naively? Especially, this was published 2 years ago. To add an OP in ONNX, it needs to be available in one framework at least.

**Comment by zldrobit** [comment]
@ebarsoum MXNet has deform conv implementation (https://mxnet.incubator.apache.org/api/python/docs/api/gluon/contrib/index.html).
Moreover, PyTorch member is considering adding deform conv (https://github.com/pytorch/pytorch/issues/28486)

**Comment by vlordier** [comment]
@ebarsoum 
Deformable Convolutions are now supported indeed in those 3 frameworks

 - [TensorFlow](https://github.com/tensorflow/addons/issues/179) https://github.com/tensorflow/addons/issues/179
- [PyTorch](https://github.com/pytorch/vision/pull/1586) https://github.com/pytorch/vision/pull/1586
- [MXNet](https://mxnet.incubator.apache.org/api/python/docs/api/gluon/contrib/index.html#mxnet.gluon.contrib.cnn.DeformableConvolution)

**Comment by vlordier** [comment]
> But to generate it in ONNX it need to be converted from someplace. If this cause a massive accuracy gain, why no framework implemented it naively? Especially, this was published 2 years ago. To add an OP in ONNX, it needs to be available in one framework at least.

@ebarsoum Deformable Convolutions are also used to process equirectangular projections of omnidirectional images by parametrising the convolution kernel

![74554848-5fb79200-4f52-11ea-9862-8f86c4a097fb](https://user-images.githubusercontent.com/5443125/75030134-62a70b00-549b-11ea-8ee0-15f9f69cdc39.png)

More recent work [Kernel Transformer Networks for Compact Spherical Convolution](https://arxiv.org/pdf/1812.03115.pdf) 

[Distortion-Aware Convolutional Filters for Dense
Prediction in Panoramic Images](http://campar.in.tum.de/pub/tateno2018eccv/tateno2018eccv.pdf
)

or [Corners for Layout: End-to-End Layout Recovery from 360 Images](https://arxiv.org/pdf/1903.08094.pdf)

**Comment by jinfagang** [comment]
@HahaOptimus I can teach you have to export and deploy DCN on tensorrt.

**Comment by phybrain** [comment]
> @HahaOptimus I can teach you have to export and deploy DCN on tensorrt.

teach me， thankyou

**Comment by q5390498** [comment]
@jinfagang can you share us how to depoly DCN on tensorrt ?
